### PR TITLE
WFLY-17056 Update JPA subsystem doc for Hibernate 6

### PR DESCRIPTION
**Pull request by scottmarlow**
Signed-off-by: Scott Marlow <smarlow@redhat.com>
https://issues.redhat.com/browse/WFLY-17056

### CODE REVIEW

File: docs/src/main/asciidoc/_developer-guide/Jakarta_Persistence_Reference_Guide.adoc [hunk 1]
```diff
@@ -698,33 +698,20 @@ Updated `jboss-deployment-structure.xml`:
...
 |=======================================================================
 |Property |Purpose
 
-|hibernate.id.new_generator_mappings =true |New applications should let
-this default to true, older applications with existing data might need
-to set to false (see note below). It really depends on whether your
-application uses the @GeneratedValue(AUTO) which will generates new key
-values for newly created entities. The application can override this
-value (in the persistence.xml).
+|hibernate.id.new_generator_mappings can no longer be set to false by applications as per Hibernate 6 change to only support hibernate.id.new_generator_mappings=true.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_developer-guide/Jakarta_Persistence_Reference_Guide.adoc:708]: This is not a 2 cell table row any more.

Also, 
s/per Hibernate 6/per the Hibernate 6/g

Comment (bstansberry) [reply docs/src/main/asciidoc/_developer-guide/Jakarta_Persistence_Reference_Guide.adoc:708]: Also, 'can' should be 'Can' -- capitalize the first word in the 'Purpose' cell.

File: docs/src/main/asciidoc/_developer-guide/Jakarta_Persistence_Reference_Guide.adoc [hunk 2]
```diff
@@ -746,40 +734,13 @@ across all application deployments on the AS instance).
...
-|hibernate.jpa.compliance.closed=true |
-
-|hibernate.jpa.compliance.query=true |
-
-|hibernate.jpa.compliance.list=true |
-
-|hibernate.jpa.compliance.caching=true |
-
-|hibernate.jpa.compliance.proxy=true |
+|hibernate.jpa.compliance=true | sets all of the various hibernate.jpa.* properties to true (hibernate.jpa.compliance.transaction, hibernate.jpa.compliance.closed,hibernate.jpa.compliance.query, hibernate.jpa.compliance.list, hibernate.jpa.compliance.caching, hibernate.jpa.compliance.proxy
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_developer-guide/Jakarta_Persistence_Reference_Guide.adoc:737]: s/sets/Sets

**Review by bstansberry — CHANGES_REQUESTED** [review]
See previous comments for requested changes.

**Comment by scottmarlow** [comment]
> See previous comments for requested changes.

Requested changes pushed, thanks!

**Comment by bstansberry** [comment]
Thanks @scottmarlow

**Pull request merged**